src/composables/queries/usePoolStakedSharesQuery.ts

```diff
@@ -0,0 +1,81 @@
+import { UseQueryOptions } from 'react-query/types';
+import { computed, reactive, Ref } from 'vue';
+import { useQuery } from 'vue-query';
+
+import QUERY_KEYS from '@/constants/queryKeys';
+import { PoolGauges } from './usePoolGaugesQuery';
+import { LiquidityGauge } from '@/services/balancer/contracts/contracts/liquidity-gauge';
+import { bnSum } from '@/lib/utils';
+import { formatUnits } from '@ethersproject/units';
+import useWeb3 from '@/services/web3/useWeb3';
+
+/**
+ * TYPES
+ */
+type QueryResponse = string;
+
+/**
+ * Fetches staked shares for a specific pool.
+ *
+ * We have to do this via the contract because the subgraph is too slow.
+ * We want users to receive instant feedback that their staked balances are updated
+ */
+export default function usePoolStakedSharesQuery(
+  poolGauges: Ref<undefined> | Ref<PoolGauges>,
```

Comment:
`Ref<PoolGauges | undefined>` doesn't work here?

Reply:
Weird, when I first tried this it was throwing an error so that's why I wrote it like this. Now it seems fine so maybe it was another issue.